### ISSUE TITLE
aot eager accuracy error: coat_lite_mini

### ISSUE THREAD

**Issue by desertfire** [pytorch/torchdynamo#1274]
Repro:
```
benchmarks/timm_models.py --accuracy-aot-nop -d cuda --float32 --only coat_lite_mini
cuda eval  coat_lite_mini                     torchdynamo.utils: [WARNING] RMSE (res-fp64): 0.61835, (ref-fp64): 0.00103
```

This is an inference error.

**Comment by anijain2305** [comment]
Repro

~~~~


import torch
import torchdynamo
from torch import tensor, device
import torch.fx as fx
from torchdynamo.testing import rand_strided
from math import inf
from torchdynamo.debug_utils import run_fwd_maybe_bwd
from torchdynamo.debug_utils import same_two_models

args = [((1, 64, 56, 56), (64, 1, 3584, 64), torch.float32, 'cuda', False)]
args = [rand_strided(sh, st, dt, dev).requires_grad_(rg) for (sh, st, dt, dev, rg) in args]


from torch.nn import *
class Repro(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.mod_serial_blocks1_0__cpe_proj = Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), groups=64)



    def forward(self, view):
        mod_serial_blocks1_0__cpe_proj = self.mod_serial_blocks1_0__cpe_proj(view)
        add = mod_serial_blocks1_0__cpe_proj + view;  mod_serial_blocks1_0__cpe_proj = view = None
        flatten_1 = add.flatten(2);  add = None
        return (flatten_1,)


mod = Repro().cuda()
opt_mod = torchdynamo.optimize("aot_eager")(mod)


mod.eval()
opt_mod.eval()

with torch.cuda.amp.autocast(enabled=False):
    assert same_two_models(mod, mod, args), "Eager itself failed"
    assert same_two_models(mod, opt_mod, args), "Dynamo failed"
    
~~~

**Comment by anijain2305** [comment]
cc @Chillee Does anything look off to you? If I remove `flatten`, the accuracy passes.

**Comment by ezyang** [comment]
This looks like reshape shenanigans. Do you regenerate contiguous inputs in the test harness? That would disagree with the non-standard contiguous input it was traced with

**Comment by ezyang** [comment]
Here's the joint graph:

```
        _reshape_alias = torch.ops.aten._reshape_alias.default(add, [1, 64, 3136], [64, 1, 64]);  add = None
        _reshape_alias_8 = torch.ops.aten._reshape_alias.default(tangents_1, [1, 64, 56, 56], [200704, 3136, 56, 1]);  tangents_1 = None
```

Notice the hard coded strides. This means that if the meta function for convolution disagrees with the real stride, this will cause an error.

If I add

```
      def forward(self, view):
          mod_serial_blocks1_0__cpe_proj = self.mod_serial_blocks1_0__cpe_proj(view)
          add = mod_serial_blocks1_0__cpe_proj + view;  mod_serial_blocks1_0__cpe_proj = view = None
          print("add stride", add.stride())
          flatten_1 = add.flatten(2);  add = None
          return (flatten_1,)
  
```

I see 

```
add stride (200704, 3136, 56, 1)
add stride (200704, 1, 3584, 64)
```

(funnily enough, the print also goes away).

@eellison this would be another good user for crossref testing.

**Comment by ezyang** [comment]
BTW, we can clue people that this sort of thing has happened in the accuracy test, by checking if the elements in one tensor are a permutation of the other (this can be done by sorting both tensors and then comparing). If that happens, it is usually a view disaster.

**Comment by ezyang** [comment]
Caveat; this is most useful if you're comparing intermediates; obviously if you have scrambled tensor and you add it to another tensor, the permutation will no longer hold.

**Comment by eellison** [comment]
Confirmed https://github.com/pytorch/pytorch/pull/82846 fixes this.

**Comment by desertfire** [comment]
> Confirmed [pytorch/pytorch#82846](https://github.com/pytorch/pytorch/pull/82846) fixes this.

Can you give other benchmarks in https://github.com/pytorch/torchdynamo/issues/1272 a try? It will help us to narrow down the list sooner. The repro instructions are in those sub-issues.
